src/test/interface/pathfinder_test.cpp

```diff
@@ -0,0 +1,160 @@
+     << " method=pathfinder"
+     << " num_paths=1";
+  run_command_output out = run_command(ss.str());
+  ASSERT_FALSE(out.hasError);
+
+  std::vector<std::string> test_result_1path = {"test", "tmp_pf.csv"};
+  std::fstream result_stream(convert_model_path(test_result_1path));
+  std::stringstream result_sstream;
+  result_sstream << result_stream.rdbuf();
+  result_stream.close();
+  std::string output = result_sstream.str();
+  EXPECT_EQ(1, count_matches("# Elapsed Time:", output));
+  EXPECT_EQ(1, count_matches(" seconds (Pathfinder)", output));
+}
+
+TEST_F(CmdStan, pathfinder_multi_good) {
+  std::stringstream ss;
+  ss << convert_model_path(multi_normal_model)
+     << " output refresh=0 file=" << convert_model_path(test_arg_output)
+     << " method=pathfinder"
+     << " num_paths=8";
+  run_command_output out = run_command(ss.str());
+  ASSERT_FALSE(out.hasError);
+  std::vector<std::string> test_result_8path = {"test", "tmp_pf_8.csv"};
+  std::fstream result_stream(convert_model_path(test_result_8path));
+  std::stringstream result_sstream;
+  result_sstream << result_stream.rdbuf();
+  result_stream.close();
+  std::string output = result_sstream.str();
+  EXPECT_EQ(1, count_matches("# Elapsed Time:", output));
+  EXPECT_EQ(1, count_matches(" seconds (Pathfinder)", output));
+}
+
+TEST_F(CmdStan, pathfinder_diagnostic_json) {
+  std::stringstream ss;
+  ss << convert_model_path(multi_normal_model)
+     << " output refresh=0 file=" << convert_model_path(test_arg_output)
+     << " diagnostic_file=" << convert_model_path(test_arg_diags)
+     << " method=pathfinder";
+  run_command_output out = run_command(ss.str());
+  ASSERT_FALSE(out.hasError);
+
+  std::fstream result_stream(convert_model_path(test_result_diags));
+  std::stringstream result_sstream;
+  result_sstream << result_stream.rdbuf();
+  result_stream.close();
+  std::string output = result_sstream.str();
+  ASSERT_FALSE(output.empty());
+  rapidjson::Document document;
+  ASSERT_FALSE(document.Parse<0>(output.c_str()).HasParseError());
+}
+
+TEST_F(CmdStan, pathfinder_lbfgs_iterations) {
+  std::stringstream ss;
+  ss << convert_model_path(eight_schools_model)
+     << " data file=" << convert_model_path(eight_schools_data)
+     << " output refresh=0 file=" << convert_model_path(test_arg_output)
+     << " diagnostic_file=" << convert_model_path(test_arg_diags)
+     << " method=pathfinder max_lbfgs_iters=10";
+  run_command_output out = run_command(ss.str());
+  ASSERT_FALSE(out.hasError);
+
+  std::fstream result_stream(convert_model_path(test_result_diags));
+  std::stringstream result_sstream;
+  result_sstream << result_stream.rdbuf();
+  result_stream.close();
+  std::string output = result_sstream.str();
+  ASSERT_FALSE(output.empty());
+  rapidjson::Document document;
+  ASSERT_FALSE(document.Parse<0>(output.c_str()).HasParseError());
+  EXPECT_EQ(1, count_matches("\"10\" : {\"iter\" : 10", output));
```

Comment:
This appears to be failing

Reply:
tweaked test logic - should reliably pass.